compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -358,16 +357,6 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
-    fn check_expr_box(&self, expr: &'tcx hir::Expr<'tcx>, expected: Expectation<'tcx>) -> Ty<'tcx> {
-        let expected_inner = expected.to_option(self).map_or(NoExpectation, |ty| match ty.kind() {
-            ty::Adt(def, _) if def.is_box() => Expectation::rvalue_hint(self, ty.boxed_ty()),
-            _ => NoExpectation,
-        });
-        let referent_ty = self.check_expr_with_expectation(expr, expected_inner);
-        self.require_type_is_sized(referent_ty, expr.span, traits::SizedBoxType);
```

Comment:
Weird, my comment from the commit didn't copy over to the pr.

I think you can remove `ObligationCauseCode::SizedBoxType`, it probably(?) isn't used anywhere else.

Reply:
Removed that and `ExprPrecedence::Box` which is no longer used